pr_agent/tools/pr_extended_code_suggestions.py

```diff
@@ -0,0 +1,209 @@
+        environment = Environment(undefined=StrictUndefined)
+        system_prompt = environment.from_string(get_settings().pr_code_suggestions_prompt.system).render(variables)
+        user_prompt = environment.from_string(get_settings().pr_code_suggestions_prompt.user).render(variables)
+        if get_settings().config.verbosity_level >= 2:
+            logging.info(f"\nSystem prompt:\n{system_prompt}")
+            logging.info(f"\nUser prompt:\n{user_prompt}")
+        response, finish_reason = await self.ai_handler.chat_completion(model=model, temperature=0.2,
+                                                                        system=system_prompt, user=user_prompt)
+
+        return response
+
+    def _prepare_pr_code_suggestions(self) -> str:
+        review = self.prediction.strip()
+        try:
+            data = json.loads(review)
+        except json.decoder.JSONDecodeError:
+            if get_settings().config.verbosity_level >= 1:
+                logging.info(f"Could not parse json response: {review}")
+            data = try_fix_json(review, code_suggestions=True)
+        return data
+
+    def push_inline_code_suggestions(self, data):
+        code_suggestions = []
+
+        if not data['Code suggestions']:
+            return self.git_provider.publish_comment('No suggestions found to improve this PR.')
+
+        for d in data['Code suggestions']:
+            try:
+                if get_settings().config.verbosity_level >= 1:
+                    logging.info(f"suggestion: {d}")
+                relevant_file = d['relevant file'].strip()
+                relevant_lines_str = d['relevant lines'].strip()
+                if ',' in relevant_lines_str:  # handling 'relevant lines': '181, 190' or '178-184, 188-194'
+                    relevant_lines_str = relevant_lines_str.split(',')[0]
+                relevant_lines_start = int(relevant_lines_str.split('-')[0])  # absolute position
+                relevant_lines_end = int(relevant_lines_str.split('-')[-1])
+                content = d['suggestion content']
+                new_code_snippet = d['improved code']
+
+                if new_code_snippet:
+                    new_code_snippet = self.dedent_code(relevant_file, relevant_lines_start, new_code_snippet)
+
+                body = f"**Suggestion:** {content}\n```suggestion\n" + new_code_snippet + "\n```"
+                code_suggestions.append({'body': body, 'relevant_file': relevant_file,
+                                         'relevant_lines_start': relevant_lines_start,
+                                         'relevant_lines_end': relevant_lines_end})
+            except Exception:
+                if get_settings().config.verbosity_level >= 1:
+                    logging.info(f"Could not parse suggestion: {d}")
+
+        self.git_provider.publish_code_suggestions(code_suggestions)
+
+    def dedent_code(self, relevant_file, relevant_lines_start, new_code_snippet):
```

Comment:
**Suggestion:** Consider adding a docstring to the `dedent_code` method to explain its purpose, arguments, and return value.
```suggestion
    def dedent_code(self, relevant_file, relevant_lines_start, new_code_snippet):
            """
            Dedents the code snippet.

            Args:
                relevant_file (str): The relevant file.
                relevant_lines_start (int): The start line of the relevant lines.
                new_code_snippet (str): The new code snippet.

            Returns:
                str: The dedented code snippet.
            """
```